task2/js/script.js

```diff
@@ -0,0 +1,125 @@
+function isArray(something) {
+  return something instanceof Array;
+}
+
+function isArray2(something) {
+  return Object.prototype.toString.call(something) === '[object Array]';
+}
+
+function range(from, to, step) {
+  var retArray = [];
+  var i;
+
+  if (from == undefined)
+    return retArray;
+
+  if (step == undefined)
+    step = 1;
+
+  if (to == undefined) {
+    to = from;
+    from = 0;
+  }
+
+  for (i = from; i < to; i += step)
+    retArray.push(i);
+
+  return retArray;
+}
+
+function compact(someArray) {
```

Comment:
mh, as I see it works in the next way: compact([1,undefined, null, '2', {}]) -> [true, true, true, true, true]
but it should return [1, '2', {}]